libsrc/apple2/ser/a2.ssc.s

```diff
@@ -210,6 +193,48 @@ SER_CLOSE:
         stx     Index           ; Mark port as closed
         rts
 
+;----------------------------------------------------------------------------
+; Internal function to test whether we're really talking with a 6551,
+; or at least it really looks like it.
+; Input: X= Index
+; Output: carry set if this is very probably a 6551, clear if not
```

Comment:
In 6502 asm it's pretty common to use the carry flag as an error indicator. So I suggest to revert the logic here.